bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryDirectDataWriterHelper.java

```diff
@@ -63,6 +68,10 @@ public class BigQueryDirectDataWriterHelper {
   private long appendRequestSizeBytes = 0; // number of bytes waiting for the next append request
   private long writeStreamRowCount = 0; // total offset / rows of the current write-stream
 
+  private final ExecutorService appendRowsExecutor = Executors.newSingleThreadExecutor();
```

Comment:
We might need multiple threads to see performance improvement right?

Reply:
My write benchmark tests indicate no significant variation between a single thread and multiple threads. Hence staying with a single thread here.